lib/moneta/pool.rb

```diff
@@ -298,6 +298,18 @@ def initialize(options = {}, &block)
     # by the manager after the ttl
     def close; end
 
+    def each_key(&block)
+      wrap(:each_key) do
+        raise NotImplementedError, "each_key is not supported on this proxy" \
+          unless supports? :each_key
+
+        return enum_for(:each_key) { adapter ? adapter.each_key.size : check_out! { adapter.each_key.size } } unless block_given?
+
+        adapter.each_key(&block)
+        self
+      end
+    end
```

Comment:
The pool specs were apparently already passing with each key being tested.  What does this add?

Reply:
I think they were until I fix the proxy for the transformers. The fact that I have to checkout kinda makes sense, because I added that so we have the count optimisation for the enumerator.

I think that if we do not want the "eager loaded" "count" to be passed in, then I could remove this code?

Reply:
Ok, so if I want to "honor" the "eager load" of the count that the enum_for allows to pass as a block for each adapter that implements it, then when the wrapper (e.g. pool) code runs, that block runs on the scope of the Proxy class and it won't checkout for an adapter.

So I reverted that change in proxy and for now we just keep it like that, is that ok? Or do you prefer me to leave all this code chain here and honor the passed in count for every adapter that implements that?